### PR TITLE
Implement Element::hasAttributes

### LINKED ISSUE #10748: Implement Element::hasAttributes
Blocks #2185.

* Spec: <https://dom.spec.whatwg.org/#dom-element-hasattributes>.
* Code: `components/script/dom/element.rs`, `components/script/dom/webidls/Element.webidl`.
* Test: to be written (at `dom/nodes/Element-hasAttributes.html`).

### PR DESCRIPTION
**Pull request by canova**
Fixes #10748 .
Implement Element::hasAttributes. I'm not sure if tests are enough. I'm open to suggestion :)

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10762)

### CODE REVIEW

File: tests/wpt/web-platform-tests/dom/nodes/Element-hasAttributes.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    var pWithCustomAttr = document.getElementsByTagName('p')[0];
+    assert_equals(pWithCustomAttr.hasAttributes(), true, 'hasAttributes() on element with custom attribute must return true.');
+
+    var divWithCustomAttr = document.createElement('div');
+    divWithCustomAttr.setAttribute('data-custom', 'foo');
+    assert_equals(divWithCustomAttr.hasAttributes(), true, 'hasAttributes() on dynamically created element with custom attribute must return true.');
+
+}, 'element.hasAttributes() must return true when the element has attribute.');
+
+</script>
```
Comment (frewsxcv) [inline tests/wpt/web-platform-tests/dom/nodes/Element-hasAttributes.html]: Trailing whitespace on this line

File: components/script/dom/element.rs [hunk 2]
```diff
@@ -1335,6 +1335,11 @@ impl ElementMethods for Element {
         self.attr_list.or_init(|| NamedNodeMap::new(&window_from_node(self), self))
     }
 
+    // https://dom.spec.whatwg.org/#dom-element-hasattributes
+    fn HasAttributes(&self) -> bool {
+        self.attrs.borrow().len() > 0
```
Comment (frewsxcv) [inline components/script/dom/element.rs]: Alternatively:

```rust
!self.attrs.borrow().is_empty()
```

**Comment by canaltinova** [comment]
@frewsxcv changed, thanks!

**Comment by nox** [comment]
There should be failing tests when running `tests/wpt/web-platform-tests/dom/interfaces.html` and `tests/wpt/web-platform-tests/html/dom/interfaces.html`. Please update the corresponding test expectations:

```
./mach test-wpt --log-raw raw.log /dom/interfaces.html /html/dom/interfaces.html
./mach update-wpt --no-patch raw.log
```

-S-awaiting-review +S-needs-code-changes 

---

Reviewed 4 of 4 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10762#-:-KFsF6Q2h84HGFiosrv9:1122047606)*

**Comment by canaltinova** [comment]
@nox Updated, thanks for help :)

**Comment by nox** [comment]
Do you need help rebasing it?

**Comment by canaltinova** [comment]
Sorry for delay. Rebased it, thanks!

**Comment by nox** [comment]
No problem, that's a very short delay. :)

@bors-servo r+

**Pull request merged**